### PR TITLE
{AKS} add support to create cluster with kube-proxy configuration

### PR DESCRIPTION
**Pull request by mattstam**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

`az aks create --kube-proxy-config file.json`

### General Guidelines

- [X] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [X] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Review by FumingZhang — COMMENTED** [review]
Please the lint issues scanned by CI.

Also need to rebase from main (including this [change](https://github.com/Azure/azure-cli-extensions/pull/5431)) to fix failures in `Integration Tests, Build Tests...`.

File: src/aks-preview/azcli_aks_live_test/configs/ext_matrix_default.json [hunk 1]
```diff
@@ -24,7 +24,8 @@
             "test_aks_custom_ca_trust_flow",
             "test_aks_create_with_csi_driver_v2",
             "test_aks_create_and_update_csi_driver_to_v2",
-            "test_aks_nodepool_abort"
+            "test_aks_nodepool_abort",
+            "test_aks_create_with_kube_proxy_config"
```
[the remark this answers is not in the document]
Comment (mattstam) [reply src/aks-preview/azcli_aks_live_test/configs/ext_matrix_default.json:28]: removed, thanks!

**Comment by mattstam** [comment]
> Please the lint issues scanned by CI.
> 
> Also need to rebase from main (including this [change](https://github.com/Azure/azure-cli-extensions/pull/5431)) to fix failures in `Integration Tests, Build Tests...`.

Rebased with main.

Locally I see some lint issue but they are not related to any of my changes from what I can tell, I think main just has lint issues in it?

**Comment by FumingZhang** [comment]
> > Please the lint issues scanned by CI.
> > Also need to rebase from main (including this [change](https://github.com/Azure/azure-cli-extensions/pull/5431)) to fix failures in `Integration Tests, Build Tests...`.
> 
> Rebased with main.
> 
> Locally I see some lint issue but they are not related to any of my changes from what I can tell, I think main just has lint issues in it?

Yeah, if the CI could pass, that would be fine.

**Comment by mattstam** [comment]
> Azure.azure-cli-extensions (CLI Linter on Modified Extensions)

[azure-cli-extensions (CLI Linter on Modified Extensions) ](https://github.com/Azure/azure-cli-extensions/pull/5427/checks?check_run_id=8779180589) would be the relevant CI here, right? That seems to pass. If there is any other changes needed, please let me know.

**Comment by FumingZhang** [comment]
Hey @mattstam, there are some errors in `Azure.azure-cli-extensions (Static Analysis)` need to be fixed.

> /mnt/vss/_work/1/s/src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:92:1: E302 expected 2 blank lines, found 1
> /mnt/vss/_work/1/s/src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:397:1: W293 blank line contains whitespace
> /mnt/vss/_work/1/s/src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2392:45: E221 multiple spaces before operator

Besides, there's another [NRP](https://github.com/Azure/azure-cli/pull/24071) API version update in official CLI, which leads to the errors in `Integration Tests, Build Tests...`, fixing this in PR []().

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**